modules/sc-mesh-secure-deployment/src/nats/src/cbma_adaptation.py

```diff
@@ -479,8 +571,9 @@ def __create_mac(self, randomized: bool = False, interface_mac: str = "") -> str
             octets = interface_mac.split(":")
             first_octet = int(octets[0], 16)
             flipped_first_octet = first_octet ^ 2
-
-            return f"{flipped_first_octet}:{':'.join(octets[1:])}"
+            # Format the flipped first octet back to hexadecimal
+            octets[0] = format(flipped_first_octet, "02x")
+            return ":".join(octets)
```

Comment:
More beautiful could be like in CBMA :love_letter: 
```python
mac_bytes = bytearray.fromhex(mac.replace(':', ''))
mac_bytes[0] ^= 0x2
return mac_bytes.hex(sep=':', bytes_per_sep=1)
```

Reply:
Same approach applied via commit f9396116ed139d0f41600f129f0a1b8bb7487e94.  I have to admit I don't fully understand how that locally administered bit is supposed to be used as this flipping zeroes it in case it was already set in original mac. But at least this implementation is now similar in CBMA adaptation/controller as well as Seb's scripts.